### PR TITLE
[CALCITE] Fixed MERGE INTO with INSERT statement unparsing in Big Query

### PR DESCRIPTION
**Pull request by AndrewPochapsky**
**Original Big Query unparsing:** 
```
MERGE INTO T1 AS A
USING T2 AS B
ON A.X = B.X
WHEN MATCHED THEN UPDATE SET
    Y = B.Y
WHEN NOT MATCHED THEN INSERT (X, Y) (VALUES  (B.X, B.Y))
```

**New Big Query unparsing:**
```
MERGE INTO T1 AS A
USING T2 AS B
ON A.X = B.X
WHEN MATCHED THEN UPDATE SET
    Y = B.Y
WHEN NOT MATCHED THEN INSERT (X, Y) VALUES  (B.X, B.Y)
```

### CODE REVIEW

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java [hunk 1]
```diff
@@ -539,4 +539,17 @@ private void checkParseInfixCast(String sqlType) {
     final String expected = "SELECT MOD(`FOO`, `BAR`)";
     sql(sql).ok(expected);
   }
+
+  @Test public void testMergeIntoBigQuery() {
```
Comment (dasch-1) [inline babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:543]: We should add a test verifying that the old unparsing logic still works on the same example query when not in BigQuery SQL dialect.

Comment (AndrewPochapsky) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:543]: I'll add the test. About the main comment: the only way that I can think of for now is putting if statements into the original file that checks the dialect. For this specific change that might actually be alright as it is quite a simple difference between the two. I can also spend a bit more time trying to explore different solutions.

Comment (dasch-1) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:543]: One idea: can we leave the merge unparsing logic where it is today, but move the insert logic only into the dialect files?

Comment (AndrewPochapsky) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:543]: Do you mean something like writer.getDialect().unparseInsertStatement(..) in which the default logic will be as before and BigQuery specific will be my change?

Comment (dasch-1) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:543]: Yes, exactly.

**Pull request merged**